### DISCUSSION TITLE
Get type of all routes to validate

### DISCUSSION THREAD

**Discussion by moishinetzer** [sandulat/routes-gen#22] [Q&A] [answered]
Is there a way of getting all the possible options of my routes all as a type so I could do the following:

```ts
const location: CheckAllMyRoutes = "/contact"
<Link to={location} />
```

Or even with dynamic routes it will understand to ignore the dynamic part, e.g.

```ts
const location: CheckAllMyRoutes = "/business/custom-business"
```

So that I can validate to make sure the location is always a correct route.

**Comment by sandulat** [comment, marked as answer]
Hey @moishinetzer! Here's an example:

```ts
import { RouteParams } from "routes-gen";

const location: keyof RouteParams = "/contact";
```

Speaking of dynamic parts, just use the `route` function directly, since it already returns an instance of `RouteParams`, regardless of the value of dynamic parts. Here's an example:

```ts
import { route, RouteParams } from "routes-gen";

const location: keyof RouteParams = route("/business/:business", {
  business: "custom-business",
});

// location = "/business/custom-business", and also respects the proper typings.
```

**Reply by moishinetzer** [reply]
Is there a way of validating the custom route without using the route() function? 

I'm doing some client side validation on routes sent from the server...

**Reply by sandulat** [reply]
@moishinetzer what exactly are you trying to achieve? Run-time strings validation? I'm afraid you can't validate this stuff in run-time, since there are no Typescript typings in run-time.

**Reply by moishinetzer** [reply]
Sure I see what you are saying. 

My use case is getting a route from prisma and typing that however I guess I wouldn't really need to... I just like typing as more of a documentation so I know this isnt a string its a route.

**Reply by sandulat** [reply]
Yep, you will not be able to verify that the actual value in your DB is a proper route. However, casting it to `keyof RouteParams` will give you at least something.
